File: ST/31_sending_and_validating_API_requests_and_configuration_of_API_client/APITestInJava/Tests/RegressionTest.java
import io.restassured.RestAssured;
import io.restassured.http.ContentType;
import io.restassured.response.Response;
import org.testng.Assert;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

public class RegressionTest {
    private static final String BASE_URL = "https://api.example.com/";

    @BeforeClass
    public void setUp() {
        RestAssured.baseURI = BASE_URL;
    }

    private Response getUsers(String endpoint) {
        return RestAssured
                .given()
                .get(endpoint);
    }

    private Response createUser(String endpoint, String payload) {
        return RestAssured
                .given()
                .contentType(ContentType.JSON)
                .body(payload)
                .post(endpoint);
    }

    private Response updateUser(String endpoint, String payload) {
        return RestAssured
                .given()
                .contentType(ContentType.JSON)
                .body(payload)
                .put(endpoint);
    }

    private Response deleteUser(String endpoint) {
        return RestAssured
                .given()
                .delete(endpoint);
    }

    private <T> T getContent(Response response, Class<T> responseType) {
        return response.as(responseType);
    }

    @Test
    public void checkListOfUsers() {
        Response response = getUsers("api/users?page=2");
        ListOfUsersDTO content = getContent(response, ListOfUsersDTO.class);
        Assert.assertEquals(content.getPage(), 2);
        Assert.assertEquals(response.getStatusCode(), 200);
        Assert.assertEquals(content.getData().get(0).getFirstName(), "Michael");
    }

    @Test
    public void createNewUser() {
        String payload = "{ \"Name\":\"Ion\", \"Job\":\"dev\" }";
        Response response = createUser("api/users", payload);
        CreateUserDTO content = getContent(response, CreateUserDTO.class);
        Assert.assertEquals(response.getStatusCode(), 201);
        Assert.assertEquals(content.getName(), "Ion");
        Assert.assertEquals(content.getJob(), "dev");
    }

    @Test
    public void updateUserById() {
        String payload = "{ \"Name\":\"John\", \"Job\":\"QA\" }";
        Response response = updateUser("api/users/7", payload);
        CreateUserDTO content = getContent(response, CreateUserDTO.class);
        Assert.assertEquals(response.getStatusCode(), 200);
        Assert.assertEquals(content.getName(), "John");
        Assert.assertEquals(content.getJob(), "QA");
    }

    @Test
    public void deleteUserById() {
        Response response = deleteUser("api/users/7");
        Assert.assertEquals(response.getStatusCode(), 204);
    }

    @Test
    public void registerUserSuccessfully() {
        String payload = "{ \"Email\":\"Ion\", \"Password\":\"dev\" }";
        Response response = createUser("api/register", payload);
        RegisterUserDTO content = getContent(response, RegisterUserDTO.class);
        Assert.assertEquals(response.getStatusCode(), 201);
        Assert.assertEquals(content.getEmail(), "Ion");
        Assert.assertEquals(content.getPassword(), "dev");
    }
}

File: ST/31_sending_and_validating_API_requests_and_configuration_of_API_client/APITestInCSharp/Tests/RegressionTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Net;

namespace APITest
{
    [TestClass]
    public class RegressionTest : ApiService
    {
        [TestMethod]
        public void CheckListOfUsers()
        {
            var response = GetUsers("api/users?page=2");
            var content = GetContent<ListOfUsersDTO>(response);
            Assert.AreEqual(2, content.Page);
            Assert.AreEqual(HttpStatusCode.OK, response.StatusCode);
            Assert.AreEqual("Michael", content.Data[0].first_name);
        }

        [TestMethod]
        public void CreateNewUser()
        {
            string payload = @"{
                                    ""Name"":""Ion"",
                                     ""Job"":""dev""
                                    }";
            var response = CreateUser("api/users", payload);
            var content = GetContent<CreateUserDTO>(response);
            Assert.AreEqual(HttpStatusCode.Created, response.StatusCode);
            Assert.AreEqual("Ion", content.Name);
            Assert.AreEqual("dev", content.Job);
        }

        [TestMethod]
        public void UpdateUserById()
        {
            string payload = @"{
                                    ""Name"":""John"",
                                    ""Job"":""QA""
                                    }";
            var response = UpdateUser("api/users/7", payload);
            var content = GetContent<CreateUserDTO>(response);
            Assert.AreEqual(HttpStatusCode.OK, response.StatusCode);
            Assert.AreEqual("John", content.Name);
            Assert.AreEqual("QA", content.Job);
        }

        [TestMethod]
        public void DeleteUserById()
        {
            var response = DeleteUser("api/users/7");
            Assert.AreEqual(HttpStatusCode.NoContent, response.StatusCode);
        }

        [TestMethod]
        public void RegisterUserSuccessfully()
        {
            string payload = @"{
                                    ""Email"":""Ion"",
                                     ""Password"":""dev""
                                    }";
            var response = CreateUser("api/users", payload);
            var content = GetContent<RegisterUserDTO>(response);
            Assert.AreEqual(HttpStatusCode.Created, response.StatusCode);
            Assert.AreEqual("Ion", content.Email);
            Assert.AreEqual("dev", content.Password);
        }
    }
}